sql/moz-fx-data-shared-prod/subscription_platform_derived/stripe_logical_subscriptions_history_v1/query.sql

```diff
@@ -94,6 +101,27 @@ subscriptions_history_charge_summaries AS (
   LEFT JOIN
     `moz-fx-data-shared-prod.stripe_external.refund_v1` AS refunds
     ON charges.id = refunds.charge_id
+  -- cards have postal code but no state
+  LEFT JOIN
+    `moz-fx-data-shared-prod.static.us_zip_code_prefixes_v1` AS card_us_zip
+    ON cards.country = "US"
+    AND LEFT(cards.address_zip, 3) = card_us_zip.zip_code_prefix
+  LEFT JOIN
+    `moz-fx-data-shared-prod.static.ca_postal_districts_v1` AS card_ca_post
+    ON cards.country = "CA"
+    AND UPPER(LEFT(cards.address_zip, 1)) = card_ca_post.postal_district_code
+  -- charges usually have postal code and are sometimes associated with
+  -- a card that does not have a state or postal code
```

Comment:
I don't know enough about relationship between cards and charges to know if this is safe to do but I thought I'd add this here so it's easier to dicuss.  From the cards that have postal codes, they almost always match the the one in the charge.
e.g. this returns two distinct cards
```sql
SELECT 
  charge.id,
  card.id,
  charge.billing_detail_address_postal_code,
  card.address_zip,
FROM
  `moz-fx-data-shared-prod.stripe_external.charge_v1` AS charge
LEFT JOIN
  `moz-fx-data-shared-prod.stripe_external.card_v1` AS card
ON
  (card.id = charge.card_id)
WHERE
  charge.billing_detail_address_postal_code != card.address_zip
```

Reply:
Ok I didn't notice that the other table was already using charges.  I'll remove the card one